### PR TITLE
new: [misp-workflow-modules] Event threat level if logic module

### LINKED ISSUE #8978: Feature Request:
### Is your feature request related to a problem? Please describe.

We wanted to check the threat level of an event and if its high send a message to teams.

### Describe the solution you'd like

A logic IF block to check the threat level of the event thats being published.

### Describe alternatives you've considered

None at the moment

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by vincenzocaputo**
#### What does it do?

This pull request adds a workflow logic module to check the threat level of the event. 

resolves #8978

#### Questions

- [ ] Does it require a DB change?
- [ ] Are you using it in production?
- [ ] Does it require a change in the API (PyMISP for example)?

### CODE REVIEW

File: app/Model/WorkflowModules/logic/Module_threat_level_if.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    public $version = '0.1';
+    public $description = 'Threat Level IF / ELSE condition block. The `then` output will be used if the encoded conditions is satisfied, otherwise the `else` output will be used.';
+    public $icon = 'code-branch';
+    public $inputs = 1;
+    public $outputs = 2;
+    public $html_template = 'if';
+    public $expect_misp_core_format = true;
+    public $params = [];
+
+    private $Attribute, $SharingGroup;
```
Comment (mokaddem) [inline app/Model/WorkflowModules/logic/Module_threat_level_if.php]: This is unused. Can be deleted.

File: app/Model/WorkflowModules/logic/Module_threat_level_if.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    private $operators = [
+        'equals' => 'Is',
+        'not_equals' => 'Is not',
+        'greater_or_equal_than' => 'Greater or equal than',
+        'less_or_equal_than' => 'Less or equal than',
+    ];
+    private $threatlevels_mapping = [
```
Comment (mokaddem) [inline app/Model/WorkflowModules/logic/Module_threat_level_if.php]: To be future proof, I'd use this function instead
https://github.com/MISP/MISP/blob/e6001bc9fa81938e68042105f43a4002ae9455a4/app/Model/ThreatLevel.php#L26-L31
Usage on how to call it looks like this
https://github.com/MISP/MISP/blob/e6001bc9fa81938e68042105f43a4002ae9455a4/app/Controller/ServersController.php#L220

**Comment by mokaddem** [comment]
Hey,
Thanks for the great PR!
Please have a look at the comments so that we can merge it in `develop`.
Cheers!

File: app/Model/WorkflowModules/logic/Module_threat_level_if.php (hunk 2 above)
Comment (vincenzocaputo) [reply app/Model/WorkflowModules/logic/Module_threat_level_if.php]: I wasn't aware of this function, I was looking for it. Thanks!

**Comment by mokaddem** [comment]
Many thanks for your work!

**Pull request merged**